### DISCUSSION TITLE
why can't I assign to vec3<f32>?

### DISCUSSION THREAD

**Discussion by shi-yan** [gpuweb/gpuweb#3478] [Q&A] [answered]
I have the following:

```
        var color = vec4<f32>(0.0, 0.0, 0.0, 0.0);
        let c = textureSampleLevel(v_result, s_diffuse, tex_coord,0);

        color.rgb *= c.rgb;
```

this doesn't work in webgpu shader, with the error message:

```
:62:15 error: cannot assign to value of type 'vec3<f32>'
```

Why? I think this is doable in glsl?

I had to change it to:

```
        let c = textureSampleLevel(v_result, s_diffuse, tex_coord,0);

        color = vec4(color.rgb * c.rgb, 1.0);
```

**Comment by munrocket** [comment, marked as answer]
According to issue https://github.com/gpuweb/gpuweb/issues/737 this assignment will be added in Post-V1.

**Reply by jiangzhaoming** [reply]
Thanks for your answer!
To be more detailed, assignment statement requires lhs being a reference [[Spec](https://gpuweb.github.io/gpuweb/wgsl/#simple-assignment-section)], and currently only selecting a single component from a vector (e.g. `color.r` or `color[1]`) [[Spec](https://gpuweb.github.io/gpuweb/wgsl/#component-reference-from-vector-reference)] can result in a reference, selecting multiple component will result in value type rather than reference [[Spec](https://gpuweb.github.io/gpuweb/wgsl/#vector-multi-component)].
Hopefully after V1 we can enable reference for vector swizzle.
